include/open62541/plugin/accesscontrol.h

```diff
@@ -127,6 +164,13 @@ struct UA_AccessControl {
                                                       UA_DateTime endTimestamp,
                                                       bool isDeleteModified);
 #endif
+    UA_Boolean (*checkUserDatabase)(const UA_UserNameIdentityToken *userToken, UA_String *roleName);
+
+#ifdef UA_ENABLE_ROLE_PERMISSION
```

Comment:
The roles are attached to a Session.
It makes sense to return the roles from `activateSession` (also in the AccessControl plugin) and store the returned roles in the session. Looking up the roles (e.g. from an ActiveDirectory service) may take a lot of time.
So caching the roles in the Session is a must requirement.

We have an API for attaching attributes to a session.
https://github.com/open62541/open62541/blob/73a60b501094876d637c78878e0c90da34b42cf7/include/open62541/server.h#L524
Use the key "0:roles" for storing the roles in the session.

With that the following methods are not needed:
- checkUserDatabase
- setRoleAccessPermission
- readUserDefinedRolePermission
- checkTheRoleSessionLoggedIn